identity/generate.go

```diff
@@ -58,10 +59,24 @@ func GetPublicIdentity(b64Identity string) (*string, error) {
 }
 
 func UpgradeIdentity(b64Identity string) (*string, error) {
-	var identity interface{}
+	identity := orderedmap.New()
 	err := b64json.Decode(b64Identity, &identity)
 	if err != nil {
 		return nil, err
 	}
+
+	_, isPrivate := identity.Get("private_encryption_key")
+	target, found := identity.Get("target")
+	if found && target == "email" && !isPrivate {
```

Comment:
I think you can return an error `if !found`

Reply:
Good point, done.